spec/shameless/index_spec.rb

```diff
@@ -13,18 +13,38 @@ class MyModel
 
         index :foo do
           integer :my_id
+          string :check_in_date
           shard_on :my_id
         end
       end
     end
+  end
 
-    MyModel.put(primary_id: 1, my_id: 1, foo: 'bar')
+  after(:context) do
+    Object.send(:remove_const, :MyModel)
+    Object.send(:remove_const, :Store)
+  end
 
-    instance = MyModel.foo_index.where(my_id: 1).first
+  let(:today) { Date.today.to_s }
+  let(:tomorrow) { (Date.today + 1).to_s }
 
-    expect(instance[:foo]).to eq('bar')
+  it 'allows querying by a named index' do
+    MyModel.put(primary_id: 1, my_id: 1, foo: 'bar', check_in_date: today)
+    MyModel.put(primary_id: 2, my_id: 1, foo: 'baz', check_in_date: tomorrow)
 
-    Object.send(:remove_const, :MyModel)
-    Object.send(:remove_const, :Store)
+    results = MyModel.foo_index.where(my_id: 1)
+
+    expect(results.size).to eq(2)
+    expect(results.first[:foo]).to eq('bar')
+  end
+
+  it 'allows querying with block filters' do
+    MyModel.put(primary_id: 1, my_id: 1, check_in_date: today)
+    MyModel.put(primary_id: 2, my_id: 1, check_in_date: tomorrow)
+
+    results = MyModel.foo_index.where(my_id: 1) { |o| o.check_in_date > today }
```

Comment:
Exactly. So depending if you need to use local method methods variables, you'd one of those.